addons/binding/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/channel/ChannelConfiguration.java

```diff
@@ -26,10 +27,10 @@
 public class ChannelConfiguration {
 
     private final @Nullable String dpt;
-    private final GroupAddressConfiguration mainGA;
+    private final @Nullable GroupAddressConfiguration mainGA;
```

Comment:
No, that's a pretty fundamental assumption: the first one is the "main" GA - and IF there is a configuration, it MUST contain an address.

Are you maybe changing the meaning of "main GA"? Then please don't do so for the basic binding.

Actually, I don't think it is a good idea to reuse this class. Advanced binding(s) may make their own choices how a channels is configured. That's why this is in an "internal" package - it is _internal_ to the basic binding.

Reply:
mmh.. It is related to https://github.com/openhab/openhab2-addons/pull/3500#discussion_r189029968 . Maybe some of the internal classes of the basic binding should be exposed in order to facilitate re-usage elsewhere.